lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+use crate::BytesDecoder;
+
+use super::BoxedFramingError;
+use bytes::{Buf, Bytes, BytesMut};
+use derivative::Derivative;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio;
+use tokio::task::JoinHandle;
+use tokio_util::codec::Decoder;
+use tracing::{info, warn};
+use vector_config::configurable_component;
+
+const GELF_MAGIC: [u8; 2] = [0x1e, 0x0f];
+const GELF_MAX_TOTAL_CHUNKS: u8 = 128;
+const DEFAULT_CHUNKS: [Bytes; GELF_MAX_TOTAL_CHUNKS as usize] =
+    [const { Bytes::new() }; GELF_MAX_TOTAL_CHUNKS as usize];
+const DEFAULT_TIMEOUT_MILLIS: u64 = 5000;
+// TODO: ask what would be an appropriate default value for this
+const DEFAULT_PENDING_MESSAGES_LIMIT: usize = 1000;
```

Comment:
I don't know what value is appropriate here. Do you have any recommendation?

This limit was enforced so we have a memory-bounded decoder. 

The maximum UDP packet size is 65536 bytes... so with this limit, I think we have roughly 65MB of memory limit for pending messages storage.

However, the framing is agnostic of the transport protocol, so maybe other protocols does not have that per-message size limit and thus this can be "theoretically unbounded" (for example, reading raw bytes from a file).

Should we enforce too a per-message limit such as https://github.com/vectordotdev/vector/blob/1579627ee3c63347fa68fe8ab3e0105bbe6ff3d9/lib/codecs/src/decoding/framing/character_delimited.rs#L51 ?

Reply:
Addressed this in 85edb0014787b2b1e340078b2a275603b2f0f7f6. Feel free to resolve this thread if the change is what you expected